usrsctplib/netinet/sctputil.c

```diff
@@ -2529,7 +2537,21 @@ sctp_timer_stop(int t_type, struct sctp_inpcb *inp, struct sctp_tcb *stcb,
 	}
 	tmr->self = NULL;
 	tmr->stopped_from = from;
-	(void)SCTP_OS_TIMER_STOP(&tmr->timer);
+	if (SCTP_OS_TIMER_STOP(&tmr->timer)) {
+		/*
+		 * If timer was actually stopped, decrement reference counts that were
+		 * incremented in sctp_timer_start.
+		 */
+		if (inp) {
+			SCTP_INP_DECR_REF(inp);
+		}
+		if (stcb) {
+			atomic_add_int(&stcb->asoc.refcnt, -1);
+		}
+		if (net) {
+			sctp_free_remote_addr(net);
```

Comment:
Not really sure about this, but better to clarify rather than ignore: shouldn't reference be decremented in the reverse order of being incremented?

Reply:
It doesn't really make a difference here, but I could change it for stylistic purposes if people want.